src/tools/illink/src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1317,16 +1317,39 @@ protected void MarkCustomAttributeProperties(ICustomAttribute ca, TypeDefinition
 
         protected void MarkCustomAttributeProperty(CustomAttributeNamedArgument namedArgument, TypeDefinition attribute, ICustomAttribute ca, in DependencyInfo reason, MessageOrigin origin)
         {
-            PropertyDefinition? property = GetProperty(attribute, namedArgument.Name);
-            if (property != null)
-                MarkMethod(property.SetMethod, reason, origin);
+            TypeDefinition? type = attribute;
```

Comment:
This implements this for PublishTrimmed; to implement for PublishAot, the new semantic needs to be implemented here too:

https://github.com/dotnet/runtime/blob/09e6ddd5aae72dda15200e0d7dc1a9c25ad0bc47/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/CustomAttributeBasedDependencyAlgorithm.cs#L237-L278